drivers/entropy/entropy_stm32.c

```diff
@@ -613,13 +616,18 @@ static int entropy_stm32_rng_init(const struct device *dev)
 	LL_SYSCFG_VREFINT_EnableHSI48();
 #endif /* CONFIG_SOC_SERIES_STM32L0X */
 
+#if !defined(CONFIG_SOC_SERIES_STM32WBX)
```

Comment:
The use of HSEM has been especially introduced for WB, to protect against dual core concurrent access. So it looks to me like a non sense to put this hsem under compilation switch !WB

Reply:
yes, 
Keep z_stm32_hsem_lock in the drivers/entropy/entropy_stm32.c